File: src/kamisado/board/Square.java
package kamisado.board;

/**
 * 
 * @author chris
 * Illustrating some methods for a board square
 * Questions:  How to represent x and y ( for instance
 * X 1..8 and Y a..h )
 */

public class Square {
	
	private int x,y;
	
	public Square(int x, int y ) {
		
		setX(x);
		setY(y);
		
	}

	
	public int getX() {
		return x;
	}
	public void setX(int x) {
		this.x = x;
	}
	public int getY() {
		return y;
	}
	public void setY(int y) {
		this.y = y;
	}
	
	public boolean equals( Object o ) {
		
		if ( !(o instanceof Square)) return false;
		Square sq = (Square)o;
		
		return ( (sq.getX()==getX()) && (sq.getY()==getY()) );
		
	}

	public String toString() {
		return "("+x+","+y+")";
	}
	

	
}

File: web/js/square.js
function Square( x, y ) {
    
    this.x = x;
    this.y = y;
    
}

Square.prototype.getX = function() {
    return this.x;
};

Square.prototype.getY = function() {
    return this.y;
};

Square.prototype.setY = function( yy ) {
    this.y = yy;
};

Square.prototype.setX = function( xx ) {
    this.x = xx;
};

Square.prototype.equals = function( sq ) {
    return sq.getX()==getX() && sq.getY()==getY();
};

Square.prototype.toString = function () {
    return "Square("+this.getX()+","+this.getY()+")";
}



